src/openforms/prefill/base.py

```diff
@@ -129,3 +129,14 @@ def get_identifier_value(
     @classmethod
     def configuration_context(cls) -> JSONObject | None:
         return None
+
+    def verify_initial_data_ownership(self, submission: Submission) -> None:
+        """
+        Hook to check if the authenticated user is the owner of the object
+        referenced to by `initial_data_reference
+
+        :param submission: an active :class:`Submission` instance
+        """
+        raise NotImplementedError(
+            "You must implement the 'verify_initial_data_ownership' method."
+        )
```

Comment:
please put this method close to `get_prefill_values_from_options` as they belong together, and preferably put it *before* since it should be called first anyway.